SignalServiceKit/src/Util/TypingIndicators.swift

```diff
@@ -428,8 +454,11 @@ public class TypingIndicatorsImpl: NSObject, TypingIndicators {
             guard delegate.areTypingIndicatorsEnabled() else {
                 return
             }
-
-            NotificationCenter.default.postNotificationNameAsync(TypingIndicatorsImpl.typingIndicatorStateDidChange, object: recipientId)
+            guard let threadId = thread.uniqueId else {
+                owsFailDebug("Thread is missing id.")
+                return
+            }
+            NotificationCenter.default.postNotificationNameAsync(TypingIndicatorsImpl.typingIndicatorStateDidChange, object: threadId)
```

Comment:
Huh - isn't `object` observation scoping based on reference (not value) equality?